app/src/main/java/de/grobox/transportr/settings/SettingsManager.kt

```diff
@@ -92,6 +100,23 @@ class SettingsManager @Inject constructor(private val context: Context) {
             }
         }
 
+    @Throws(UnknownHostException::class, IllegalStateException::class)
+    fun getProxy(proxyPrefOverrides: Map<String, Any>): Proxy {
```

Comment:
All uses of this function seem to set at most one new setting. Why not simply using a `Pair<String, Any>` here?

Reply:
You're right but I think the code is actually more readable with the map compared to a single `Pair<String, Any>?`, if we want to handle mismatched types the same way.

Compare
```kotlin
val isEnabled = proxyPrefOverrides[PROXY_ENABLE] as Boolean? ?: settings.getBoolean(PROXY_ENABLE, false)
```
to
```kotlin
val isEnabled =
    if (proxyPrefOverride?.first == PROXY_ENABLE && proxyPrefOverride.second is Boolean)
        proxyPrefOverride.second as Boolean
    else
        settings.getBoolean(PROXY_ENABLE, false)
```
or (avoiding the explicit cast)
```kotlin
val overrideKey = proxyPrefOverride?.first
val overrideVal = proxyPrefOverride?.second
val isEnabled = if (overrideKey == PROXY_ENABLE && overrideVal is Boolean)
    overrideVal else settings.getBoolean(PROXY_ENABLE, false)
```